src/dfx/src/commands/deps/init.rs

```diff
@@ -0,0 +1,158 @@
+use crate::lib::deps::{
+    create_init_json_if_not_existed, get_pull_canisters_in_config, get_service_candid_path,
+    load_init_json, load_pulled_json, save_init_json, validate_pulled,
+};
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::util::check_candid_file;
+
+use anyhow::{anyhow, bail, Context};
+use candid::parser::types::IDLTypes;
+use candid::parser::value::IDLValue;
+use candid::types::Type;
+use candid::{IDLArgs, Principal, TypeEnv};
+use clap::Parser;
+use fn_error_context::context;
+use slog::{info, warn};
+
+/// Set init argument for a pulled canister.
+#[derive(Parser)]
+pub struct DepsInitOpts {
+    /// Name of the pulled canister (as defined in dfx.json) or its Principal.
+    /// If not specified, all pulled canisters will be set.
+    canister: Option<String>,
+
+    /// Specifies the init argument.
+    #[clap(long, requires("canister"))]
+    argument: Option<String>,
+
+    /// Specifies the data type of the init argument.
+    #[clap(long, requires("argument"), possible_values(&["idl", "raw"]))]
+    argument_type: Option<String>,
+}
+
+pub async fn exec(env: &dyn Environment, opts: DepsInitOpts) -> DfxResult {
+    let logger = env.get_logger();
+    let pull_canisters_in_config = get_pull_canisters_in_config(env)?;
+    if pull_canisters_in_config.is_empty() {
+        info!(logger, "There is no pull dependencies defined in dfx.json");
+        return Ok(());
+    }
+
+    let project_root = env.get_config_or_anyhow()?.get_project_root().to_path_buf();
+    let pulled_json = load_pulled_json(&project_root)?;
+    validate_pulled(&pulled_json, &pull_canisters_in_config)?;
+
+    create_init_json_if_not_existed(&project_root)?;
+    let mut init_json = load_init_json(&project_root)?;
+
+    match opts.canister {
+        Some(canister) => {
+            let canister_id = match pull_canisters_in_config.get(&canister) {
+                Some(canister_id) => *canister_id,
+                None => Principal::from_text(&canister).with_context(|| {
+                    "The canister is neither a valid Principal nor a name specified in dfx.json"
+                })?,
+            };
+
+            let idl_path = get_service_candid_path(canister_id)?;
+            let (env, _) = check_candid_file(&idl_path)?;
+            let candid_args = pulled_json.get_candid_args(&canister_id)?;
+            let candid_args_idl_types: IDLTypes = candid_args.parse()?;
+            let mut types = vec![];
+            for ty in candid_args_idl_types.args.iter() {
+                types.push(env.ast_to_type(ty)?);
+            }
+
+            let arguments = opts.argument.as_deref();
+            let arg_type = opts.argument_type.as_deref();
+
+            match (arguments, types.is_empty()) {
+                (Some(arg_str), false) => {
+                    if arg_type == Some("raw") {
+                        let bytes = hex::decode(arguments.unwrap_or(""))
```

Comment:
I don't see any e2e test that passes --argument-type raw

Also:
```suggestion
                        let bytes = hex::decode(arg_str)
```